### PR TITLE
Add `username` modifier for `auth_on_register` hooks

### PR DESCRIPTION
**Pull request by codeadict**
- Added the ability to modify the username from the `auth_on_register`
hook.
- Changed the `vmq_diversity` plugin to support this new modifier.
- Changed the `vmq_webhooks` plugin to support this new modifier.

### CODE REVIEW

**Comment by codeadict** [comment]
**Question:** Should `username` be added to the `reg_modifiers` type on https://github.com/vernemq/vernemq_dev/blob/master/src/auth_on_register_hook.erl and https://github.com/vernemq/vernemq_dev/blob/master/src/auth_on_register_m5_hook.erl?

File: apps/vmq_server/test/vmq_subscribe_SUITE.erl [hunk 1]
```diff
@@ -34,8 +34,7 @@ end_per_testcase(_, Config) ->
 all() ->
     [
      {group, mqttv4},
-     {group, mqttv5},
-     {group, try_private}
```
Comment (larshesel) [inline apps/vmq_server/test/vmq_subscribe_SUITE.erl]: nice catch - forgot to remove that one

**Comment by larshesel** [comment]
I think that would be perfect if it was added there as well

**Comment by larshesel** [comment]
@dergraf can you check that this was what you had in mind when opening #1246 ?

**Comment by larshesel** [comment]
closes #1246

**Pull request merged**